### PR TITLE
dedicated volunteer registration space

### PR DESCRIPTION
**Pull request by willblake01**
Add a volunteer landing page linked from the header. List information on volunteer positions. Add a form to register volunteers and persist to database. Streamlines the volunteering process and eliminates manual work saving time for the admins.

-Add "Volunteer" link to header
-Add "/volunteers" landing page
-Add positions card listing available positions
-Add volunteer registration form
-Save volunteer to database

### CODE REVIEW

File: app/controllers/volunteers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    if @volunteer.errors.messages[:email]
+      flash[:error] = @volunteer.errors.messages[:email][0]
+      redirect_to new_volunteer_path
+    elsif @volunteer.errors
+      flash[:error] = @volunteer.errors.messages
+      redirect_to new_volunteer_path
+    else
+      flash[:success] = 'Thank you for volunteering!'
+      redirect_to root_path
+    end
```
Comment (ColinW520) [inline app/controllers/volunteers_controller.rb:18]: How could you replace this logic block with a service?

File: app/views/volunteers/_form.html.haml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      .col
+        = simple_form_for @volunteer do |f|
+          .form-inputs
+            = f.association :event
+            = f.input :first_name, label: "First Name"
+            = f.input :last_name, label: "Last Name"
+            = f.input :phone, label: "Phone"
+            = f.input :email, label: "Email"
+            = f.input :position, label: "Preferred Position"
+            = f.input :time, :collection => %w[AM PM All\ Day]
```
Comment (ColinW520) [inline app/views/volunteers/_form.html.haml:13]: Prefer not to use legacy hash rocket syntax

File: config/routes.rb [hunk 3]
```diff
@@ -29,12 +29,12 @@
       resources :participants, only: [:index]
     end
   end
+  put "newsletter_signup", to: "newsletter_signups#create"
+  match "*path/newsletter_signup", to: "newsletter_signups#create", via: :put
```
Comment (ColinW520) [inline config/routes.rb:33]: These changes seem unrelated to volunteers... suggest only changing one thing at time, even if this does just feel like cleanup... Also, we just be using a resources block for this.

Comment (willblake01) [reply config/routes.rb:33]: @ColinW520 Hmm. I ended up with a merge conflict after merging those PRs last Friday. Not sure why that is coming in as an addition since it should already be in master. I'll look into what happened...
